### ISSUE TITLE
Checkbox A11y Pass

### ISSUE THREAD

**Issue by TestAccessibilityGIT** [OfficeDev/office-ui-fabric-react#6514]
Tracking all of the issues brought up with the 09/28 accessibility pass on Checkbox  control
Details on individual bugs can be found on [SharePoint](https://microsoft-my.sharepoint.com/:x:/r/personal/v-amaver_microsoft_com/_layouts/15/Doc.aspx?sourcedoc=%7B2356d1a0-406e-4214-8c3e-8dc6a558ff40%7D&action=default&uid=%7B2356D1A0-406E-4214-8C3E-8DC6A558FF40%7D&ListItemId=1406&ListId=%7B395E1ECB-5002-440E-81C5-6A090E98077C%7D&odsp=1&env=prodbubble)

- [x] Bug 1: [Accessibility][Programmatic access]: Narrator conveys inappropriate information on 'Standard checkbox' control.
- [x] Bug 2: [Accessibility][Screen reader]: Two scan mode focuses encountered on checkboxes when user first time selects them. 
- ~~Bug 3: [Accessibility][Forms and validation]: Group labels inside dropdowns are not associated with its corresponding control.~~ (Not applicable to Checkbox, should be covered in ContextualMenu pass)
- ~~Bug 4: [Accessibility][Usable]: Screen does not scroll down along with keyboard focus in back navigation.~~ (Not applicable to Checkbox, should be covered in ContextualMenu pass)
- [x] Bug 5: [Accessibility][Supporting the platform]: Checkbox button Controls are not visible on mouse focus in high contrast white/black themes.
 

Component: Checkbox
@micahgodbolt

**Comment by aneeshack4** [comment]
I spoke with Arvind from @TestAccessibilityGIT & he provided me with more details on how to repro some of these. Bugs 1 and 5 don't repro anymore. (note: I updated my machine to windows version 1809 OS build 17763.1 to get these repros because on 17124.345, what I was on before, these weren't reproing). 

Bug 1 lack of repro proof: The problem was that, when narrator focus lands on checkbox, it would narrate "Uncontroller checkbox description" which isn't requried and narrator's would also say the checkbox control name twice. As you can see in this print out of all the text that narrator speaks, it no longer has the extra uncontroller part & it may sound like it's saying the control name twice but that's just because the control on the example page has a label with the word checkbox in it. 
![image](https://user-images.githubusercontent.com/4161791/46979805-76f91d80-d087-11e8-8af8-ce087e62ea30.png)

Bug 5 lack of repro proof: checkbox button controls used to not be visible on mouse focus in high contrast but as you can see in the screenshot below they're now visible
![image](https://user-images.githubusercontent.com/4161791/46979986-11596100-d088-11e8-9266-be91ab2be98e.png)

Bugs 3 & 4 I'm not sure what "group label inside dropdowns" and "back navigation" refer to in the context of the Checkbox page on the Fabric website. Arvind @TestAccessibilityGIT said these issues were also logged for "Contextual Menu" and "Button" controls and that @micahgodbolt might be able to help identify the correct areas for these two issues. 

Bug 2: When scan mode's on and we select checkbox using space then use the down arrow to go to the next control, it should go to the next control "Implementation Examples" Arvind @TestAccessibilityGIT reported that narrator focus instead landed again on the checkbox name. For me, as you can see in this gif, it's actually "unable to move Narrator cursor", I'm trying to find out why. If anyone has any ideas, please let me know!
![image](https://user-images.githubusercontent.com/4161791/46980881-dad11580-d08a-11e8-9e41-76f3a220e7f3.png)

**Comment by TestAccessibilityGIT** [comment]
Hi Aneesha,
We cross validated bug 1 and 5, seems those are still appearing for us. Please find attached attachment for those issues.

Regards,
Arvind

**Comment by TestAccessibilityGIT** [comment]
Hey Arvind,

Apologies for the delay in response. I've been busy with a couple other tasks on my plate.

Thank you for providing me the recordings - it's clear that how they're reproing on your end. I'm able to repro the HC bug - I was missing the repro step of first navigating using the keyboarding then using mouse but I see it now and I'll work on fixing it.

But I'm still unable to get a repro of bug 1 (the uncontroller checkbox description). Are you using Edge? Til now I only ran narrator on Chrome I just tried it on Edge to see if I can repro it there but it actually just confused me even more because the blue boxes that Narrator draws around each thing it's reading out doesn't get drawn on Edge - does this happen for you also? Also you're also on windows version 1809 OS build 17763.1 on the fabric checkbox component page (https://developer.microsoft.com/en-us/fabric#/components/checkbox) right?

And mind sharing which application you used to record your screen with audio included? That's really useful & I'd love to use it in the future.

________________________________
From: Arvind Prakash (HCL America Inc)
Sent: Tuesday, October 16, 2018 4:28 PM
To: OfficeDev/office-ui-fabric-react; Aneesha Kommineni; OfficeDev/office-ui-fabric-react
Cc: Mention; Micah Godbolt
Subject: RE: [OfficeDev/office-ui-fabric-react] Checkbox A11y Pass (#6514)


Hi Aneesha,

We cross validated bug 1 and 5, seems those are still appearing for us. Please find attached attachment for those issues.



Regards,

Arvind

**Comment by TestAccessibilityGIT** [comment]
Hi Aneesha,
Please find inline.

Regards,
Arvind

**Comment by cliffkoh** [comment]
We should reevaluate this issue once #7521 is merged.

**Comment by aneeshack4** [comment]
![image](https://user-images.githubusercontent.com/4161791/51574739-f145f480-1e63-11e9-8697-f8d5909bf2a4.png) 
This was tested on windows version 1809 OS build 17763.253
As you can see in this image, Narrator does convey the right information on the "Standard checkbox" control - it repeats the word checkbox twice simply because that's the label (bug 1). And, in scan mode, there's only one focus rect around the checkbox (bug 2).

**Comment by aneeshack4** [comment]
Some more notes on Bug 2: I do see that Narrator in scan mode, if you press "space" to select a checkbox and then use the down arrows the navigate, the focus stays on the control & the focus rects go to the label instead of the checkbox below. But this seems to be how Narrator handles focus... not sure if we can do anything on our end to change this such that the focus immediately goes to the next control. And also not sure if that's the right move - users may want the option to stay on the current checkbox, not sure if we can definitively say they'll always want to move to the next one.

**Comment by TestAccessibilityGIT** [comment]
Bug_02: [Accessibility][Screen reader]: Two scan mode focuses encountered on checkboxes when user first time selects them. 
**Regression status :** Still repro. Now three scan mode focuses has provided on all check box controls. PFA. This issue does not repro with NVDA so we can mark it as usable and pass to narrator team. Please find attached video attachment.

**Bug_05:** [Accessibility][Supporting the platform]: Checkbox button Controls are not visible on mouse focus in high contrast white/black themes.
Regression status : Does not repro. Seems fixed on RS5_Prerelease(18317 OS Build 190111-1556)
[Bug_02.zip](https://github.com/OfficeDev/office-ui-fabric-react/files/2806468/Bug_02.zip)

**Comment by TestAccessibilityGIT** [comment]
- [x] Bug 2: [Accessibility][Screen reader]: Two scan mode focuses encountered on check boxes when user first time selects them. 
Bug 2 is still not fixed. 

Verified it on test environment:
OS: Version: 1903 (OS Build 18362.239) 
URL: https://fabricweb.z5.web.core.windows.net/oufr/7.19.0/#/examples/checkbox
